app/models/ext_management_system.rb

```diff
@@ -186,6 +205,49 @@ def hostname_format_valid?
 
   default_value_for :enabled, true
 
+  after_save :change_maintenance_for_child_managers, :if => proc { |ems| ems.changed_attributes.include?('enabled') }
+
+  def disable!
+    _log.info("Disabling EMS [#{name}] id [#{id}].")
+    update!(:enabled => false)
+    _log.info("Disabling EMS [#{name}] id [#{id}] successful.")
+  end
+
+  def enable!
+    _log.info("Enabling EMS [#{name}] id [#{id}].")
+    update!(:enabled => true)
+    _log.info("Enabling EMS [#{name}] id [#{id}] successful.")
+  end
+
+  # Move ems to maintenance zone and backup current one
+  def pause!
+    _log.info("Pausing EMS [#{name}] id [#{id}].")
+    update!(
+      :zone_before_pause => zone,
+      :zone              => Zone.maintenance_zone,
+      :enabled           => false
+    )
+    _log.info("Pausing EMS [#{name}] id [#{id}] successful.")
+  end
+
+  # Move ems to original zone, reschedule task/jobs/.. collected during maintenance
+  def resume!
+    _log.info("Resuming EMS [#{name}] id [#{id}].")
+
+    new_zone = if zone_before_pause.nil?
+                 zone.visible? ? zone : Zone.default_zone
+               else
+                 zone_before_pause
+               end
+
+    update!(
+      :zone_before_pause => nil,
+      :zone              => new_zone,
+      :enabled           => true
```

Comment:
Same concern here with `enable!`

Reply:
like above, enable! is used by enabling child providers (i.e. Amazon's NetworkProvider) - will be moved to `protected` by PR #18037

Reply:
https://github.com/ManageIQ/manageiq/pull/17452/files#diff-bc914c1dea04b4ba49055f79a8ab8b5aR813 is resuming the child managers